Ciac 10199 azure supporting client credentials flow (#34394)
demisto/content: Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/README.md

   3. *Device Code Flow*.

## Self-Deployed Azure App

To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal.

To add the registration, refer to the following [Microsoft article](https://learn.microsoft.com/en-us/defender-xdr/api-create-app-web?view=o365-worldwide) steps 1-8.

### Required permissions

- Azure Service Management - permission `user_impersonation` of type Delegated
- Microsoft Graph - permission `offline_access` of type Delegated

In order to add permission navigate:

Home -> App registrations -> Search for you App under 'all applications' -> API permissions -> Add permission -> search for the specific Microsoft API -> select the specific permission of type Delegated.

### Authentication Using the Authorization Code Flow (recommended)

1. In the **Authentication Type** field, select the **Authorization Code** option.
2. In the **Application ID** field, enter your Client/Application ID.
3. In the **Client Secret** field, enter your Client Secret.
4. In the **Tenant ID** field, enter your Tenant ID .
5. In the **Application redirect URI** field, enter your Application redirect URI.
6. Save the instance.

> 7. Run the `!azure-nsg-generate-login-url` command in the War Room and follow the instruction.

Reviewer: ```suggestion
7. Run the `!azure-nsg-generate-login-url` command in the War Room and follow the instructions.
```

### Authentication Using the Client Credentials Flow

1. Assign Azure roles using the Azure portal [Microsoft article](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal)

>    *Note:* At the 'Select members' section, assign the application you created before.

Reviewer: ```suggestion
   *Note:* In the *Select members* section, assign the application you created earlier.
```

2. To configure a Microsoft integration that uses this authorization flow with a self-deployed Azure application:

> 3. In the **Authentication Type** field, select the **Client Credentials** option.
> 4. In the **Application ID** field, enter your Client/Application ID.
> 5. In the **Subscription ID** field, enter your Subscription ID.
> 6. In the **Resource Group Name** field, enter you Resource Group Name.
> 7. In the **Tenant ID** field, enter your Tenant ID .
> 8. In the **Client Secret** field, enter your Client Secret.
> 9. Click **Test** to validate the URLs, token, and connection
> 10. Save the instance.

Reviewer: ```suggestion
   a. In the *Authentication Type* field, select the **Client Credentials** option.
   b. In the *Application ID* field, enter your Client/Application ID.
   c. In the *Subscription ID* field, enter your Subscription ID.
   d. In the *Resource Group Name* field, enter you Resource Group Name.
   e. In the *Tenant ID* field, enter your Tenant ID .
   f. In the *Client Secret* field, enter your Client Secret.
   g. Click **Test** to validate the URLs, token, and connection
   h. Save the instance.
```

   | Tenant ID                          |                                                                                                                                                                                                    | False |
   | Client Secret                      |                                                                                                                                                                                                    | False |
   | Application redirect URI           |                                                                                                                                                                                                    | False |
   | Authorization code                 | For user-auth mode - received from the authorization step. See Detailed Instructions \(?\) section.                                                                                                | False |

>    | Azure Managed Identities Client ID | The Managed Identities client ID for authentication - relevant only if the integration is running on Azure VM.                                                                                     |False |

Reviewer: Not sure why you edited this lines.  It will not render correctly. As far as I can tell, it was fine the way it was.

2. Search for Azure Network Security Groups.
3. Click **Add instance** to create and configure a new integration instance.

    | **Parameter**                      | **Description** | **Required** |
    |------------------------------------| --- | --- |
    | Application ID                     |  | False |
     | Default Subscription ID            | There are two options to set the specified value, either in the configuration or directly within the commands. However, setting values in both places will cause an override by the command value. | True |
    | Default Resource Group Name        |There are two options to set the specified value, either in the configuration or directly within the commands. However, setting values in both places will cause an override by the command value.  | True |
    | Azure AD endpoint                  | Azure AD endpoint associated with a national cloud. | False |
    | Trust any certificate (not secure) |  | False |
    | Use system proxy settings          |  | False |

>     | Authentication Type                | Type of authentication - can be Authorization Code flow \(recommended\), Device Code Flow, or Azure Managed Identities. | True |

Reviewer 2: Add to the options`Client Credentials`

There is no context output for this command.

#### Command Example

```!azure-nsg-auth-start ```

#### Human Readable Output

> To sign in, use a web browser to open the page https://microsoft.com/devicelogin
> and enter the code CODECODE to authenticate.

> > Run the !azure-nsg-auth-complete command in the War Room.

Reviewer: ```suggestion
> Run the ***!azure-nsg-auth-complete*** command in the War Room.
```

| AzureNSG.Rule.destinationPortRanges    | String   | For multiple ports, a list of destination ports. Note that for single ports, \`destinationPortRange\` will appear instead.              | 
| AzureNSG.Rule.sourceAddressPrefix      | String   | The source address.                                                                                                                     | 
| AzureNSG.Rule.destinationAddressPrefix | String   | The destination address.                                                                                                                | 
| AzureNSG.Rule.access                   | String   | The rule's access. Can be "Allow" or "Deny".                                                                                            | 
| AzureNSG.Rule.priority                 | Number   | The rule's priority. Can be from 100 to 4096.                                                                                           | 
| AzureNSG.Rule.direction                | String   | The rule's direction. Can be "Inbound" or "Outbound".                                                                                   | 

### azure-nsg-generate-login-url

***

> Generate the login url used for Authorization code flow or Client Credentials flow.

Reviewer 2: please revert :)
the command is relevant only for Authorization code flow
